pkg/kubelet/config/config_test.go

```diff
@@ -413,3 +415,27 @@ func TestPodUpdateLabels(t *testing.T) {
 	expectPodUpdate(t, ch, CreatePodUpdate(kubetypes.UPDATE, TestSource, pod))
 
 }
+
+func TestPodConfigRace(t *testing.T) {
+	eventBroadcaster := record.NewBroadcaster()
+	config := NewPodConfig(PodConfigNotificationIncremental, eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "kubelet"}))
+	seenSources := sets.NewString(TestSource)
+	var wg sync.WaitGroup
+	const iterations = 100
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		for i := 0; i < iterations; i++ {
+			config.Channel(strconv.Itoa(i))
+		}
```

Comment:
```suggestion
		ctx, cancel := context.WithCancel(context.Background())
		defer cancel()
		defer wg.Done()
		for i := 0; i < iterations; i++ {
			config.Channel(ctx, strconv.Itoa(i))
		}
```

You may not rebase on the current master branch. https://github.com/kubernetes/kubernetes/pull/107163 was merged in Jan.
/hold
after update, you can unhold.